distribution/lib/Standard/Base/0.0.0-dev/src/Data/Ordering.enso

```diff
@@ -23,42 +23,38 @@ from project.Data.Boolean import all
 
     Should there be a need to redefine the default implementation, here is a way:
     Define conversion function `Comparable.from` for your `type` and return pointer to
-    another `type` that satisfies either of the following two definitions:
+    another `type` that satisfies the following definition:
 
     ```
-    type Ordered_Comparator T
-        is_ordered = True
+    type Comparator T
         compare : T -> T -> (Ordering|Nothing)
         hash : T -> Integer
-
-    type Unordered_Comparator T
-        is_ordered = False
-        equals : T -> T -> (Boolean|Nothing)
-        hash : T -> Integer
     ```
 
-    Or `Incomparable` in case that the type `T` should not be compared at all.
-
     Every type must provide exactly one comparator, i.e., any method of form
     `Comparable.from (_:My_Type)` must always return the same comparator type.
-    Note that if there is an implicit `Default_Unordered_Comparator` assigned
-    for every type by default.
+    Note that there is an implicit `Default_Comparator` assigned for every
+    type by default.
+
+    A comparator has to implement `compare` and `hash` methods. `compare x y` method
+    returns `Ordering.Less` if x is ordered before y, `Ordering.Equal` if x is equal
+    to y, `Ordering.Greater` if x is ordered after y, and `Nothing` if x and y are
+    _incomparable_ or not equal.
 
-    Note that there has to be `is_ordered` method defined which returns a Boolean
-    indicating that the comparator is ordered. This is currently needed as there is
-    no way to define interfaces in Enso.
+    Note that _incomparable_ and _not equal_ is not, strictly speaking, the same
+    relation. But for the sake of simplicity of the `compare` method, they are the
+    same relations.
```

Comment:
I'd remove the latter paragraph, IMO it is a bit hard to undersand.

I'd phrase this as following:
```
    A comparator has to implement `compare` and `hash` methods. `compare x y` method
    returns `Ordering.Less` if x is ordered before y, `Ordering.Equal` if x is equal
    to y, `Ordering.Greater` if x is ordered after y, and `Nothing` if x and y are
    not equal but their relative ordering cannot be determined.
```

What do you think?

Reply:
Indeed _incomparable_ is a bit different from not-equal, but in Enso IMO we don't have a true incomparable, we just have not-equal which may or may not specify relative ordering between not-equal elements. There is no way of saying "I don't know if these two elements are equal or not", I can only say they are equal or not equal (+- ordering).

Reply:
Yes, I agree. That paragraph is difficult to understand. I will remove it and rewrite the former one.

Reply:
Thanks!